app/src/marketplace/components/NewProductPage.jsx

```diff
@@ -0,0 +1,63 @@
+// @flow
+
+import React, { useEffect, useState } from 'react'
+import { type Location } from 'react-router-dom'
+import styled from 'styled-components'
+import qs from 'query-string'
+import { useDispatch } from 'react-redux'
+import { replace } from 'connected-react-router'
+import { postEmptyProduct } from '$mp/modules/deprecated/editProduct/services'
+import LoadingIndicator from '$userpages/components/LoadingIndicator'
+import routes from '$routes'
+import { type ProductType } from '$mp/flowtype/product-types'
+import useIsMounted from '$shared/hooks/useIsMounted'
+import { productTypes } from '$mp/utils/constants'
+
+type Props = {
+    className?: ?string,
+    location: Location,
+}
+
+const sanitizedType = (type: ?string): ProductType => (
+    productTypes[(type || '').toUpperCase()] || productTypes.NORMAL
+)
+
+const UnstyledNewProductPage = ({ className, location: { search } }: Props) => {
+    const dispatch = useDispatch()
+
+    const isMounted = useIsMounted()
+
+    const [error, setError] = useState()
+
+    useEffect(() => {
+        const { type } = qs.parse(search)
```

Comment:
would explicit handling for no/unknown type be useful?

Reply:
Well, the "create product modal" can't be linked to. That'd be my idea: no/unknows => open the selection tool. Right now we default no/unknown to `NORMAL`. Any ideas?

Reply:
I'mma roll with that. We can handle the edge cases later if needed.